### DISCUSSION TITLE
how to add helix editor to git for windows?

### DISCUSSION THREAD

**Discussion by soloturn** [git-for-windows/git#4353] [Ideas]
vim as a default editor is always a little difficult for windows command line users, nano is a little simplistic, so i wondered if it would be possble to add helix editor instead or additionally to vim to git for windows? and if yes, which repository this would be done in?
https://packages.msys2.org/package/mingw-w64-x86_64-helix?repo=mingw64

**Comment by dscho** [comment]
This is the _first_ time I hear about the "helix" editor. As such, I doubt that there is any strong demand outside of your request to support this editor, specifically, and given that the [`.zip` file weighs in with a whopping 14.8MB](https://github.com/helix-editor/helix/releases/tag/22.12) I am strongly opposed to _including_ the helix editor in Git for Windows' installer. Even taking LZMA compression into account, the increase in installer size definitely would not be justified by the benefit.

Having said that, if you want to _use_ the helix editor, it is relatively easy: In the Git for Windows installer's page about editors, select the last option and specify a command-line to execute `hx.exe`:

![image](https://user-images.githubusercontent.com/127790/226325917-15c8d3ff-e823-4304-90ba-caed1458863d.png) 

You will note that I prefixed the command-line `winpty ~/Desktop/helix-22.12-x86_64-windows/hx.exe` with `winpty`. The reason is that I use MinTTY as terminal window for Git Bash (which is the default) and `hx.exe` has no idea how to deal with that terminal window, hence the need for `winpty` (as pointed out in Git for Windows' [release notes](https://github.com/git-for-windows/build-extra/blob/main/ReleaseNotes.md#known-issues)).

**Reply by soloturn** [reply]
@dscho, only one executable file would be necessary, the minimum size would be:
```
❯ ls -lh /usr/lib/helix/hx
16M 20. Feb 16:09 /usr/lib/helix/hx*
❯ ls -lh hx.zip
6.0M 20. Mär 15:14 hx.zip
```
which is 3 times vim. this would be too much as well, correct?

**Reply by dscho** [reply]
> this would be too much as well, correct?

Yes.

**Reply by soloturn** [reply]
i like helix because it is nearly vi, contrary to vi, hx shows what it deletes and changes, hx can be made that backspace deletes a char. and with both, ordinary windows users are not so much lost as they are with vim. but i understand - who is able to learn programming, is able to learn configure git :) :)
